system/jlib/jsocket.cpp

```diff
@@ -4691,6 +4688,35 @@ class CSocketEpollThread: public CSocketBaseThread
         }
     }
 
+    void delSelItem(SelectItem *si)
+    {
+        epoll_op(epfd, EPOLL_CTL_DEL, si, 0);
+        // Release/dtors should not throw but leaving try/catch here until all paths checked
+        try
+        {
+            si->nfy->Release();
+            si->sock->Release();
+            delete si;
+        }
+        catch (IException *e)
+        {
+            EXCLOG(e,"CSocketEpollThread::delSelItem()");
+            e->Release();
+        }
+    }
+
+    void delSelItemPos(SelectItem *si, unsigned pos)
+    {
+        unsigned last = items.ordinality();
+        if ( (last == 0) || (pos >= last) )
+            return;
```

Comment:
minor: I would say these internal functions that are here to common up, you explicitly know that the parameters are in range before you call them and this extra checking is over the top/unnecessary. At most I'd put them in a dbgassertex in case called in any other contexts in the future.

Reply:
ok, removed